addons/pos_restaurant/static/src/js/multiprint.js

```diff
@@ -282,14 +287,37 @@ const PosResMultiprintOrder = (Order) => class PosResMultiprintOrder extends Ord
         }
         return false;
     }
-    export_as_JSON(){
-        var json = super.export_as_JSON(...arguments);
-        json.multiprint_resume = JSON.stringify(this.saved_resume);
-        return json;
-    }
-    init_from_JSON(json){
-        super.init_from_JSON(...arguments);
-        this.saved_resume = json.multiprint_resume && JSON.parse(json.multiprint_resume);
+    async printChanges(){
+        let isPrintSuccessful = true;
+        const d = new Date();
+        let hours = '' + d.getHours();
+        hours = hours.length < 2 ? ('0' + hours) : hours;
+        let minutes = '' + d.getMinutes();
+        minutes = minutes.length < 2 ? ('0' + minutes) : minutes;
+
+
+        for (const printer of this.pos.unwatched.printers) {
+            const changes = this._getPrintingCategoriesChanges(printer.config.product_categories_ids);
+            if (changes['new'].length > 0 || changes['cancelled'].length > 0) {
+                const printingChanges = {
+                    new: changes['new'],
+                    cancelled: changes['cancelled'],
+                    table_name: this.pos.config.iface_floorplan ? this.getTable().name : false,
+                    floor_name: this.pos.config.iface_floorplan ? this.getTable().floor.name : false,
+                    name: this.name || 'unknown order',
+                    time: {
+                        hours,
+                        minutes,
+                    },
```

Comment:
When one of the printers failed to print, we consider it an unsuccessful print, thus, the user will have to click the submit button again. However, for some printers, there actually is no "change" but they will print again. We can't change this, those printers has to reprint. But we can let the receiver of the printed "changes" that the newly printed receipt is a "reprint" by making sure the time is the same with the previous receipt.

So, my suggestion is to _not_ derive `time` here in `printChanges` but include it when computing `printingChanges`.